proj/src/components/projects/EditProject.js

```diff
@@ -59,25 +64,62 @@ const CloseButton = styled.button`
 
 const EditProject = (props) => {
   
-  const submit = (e) => {
+  const submitProject = (e) => {
     e.preventDefault();
-    props.onSubmitProject(e.target.elements.name.value, props.projectId);
+    const inputValue = e.target.elements[0].value
+    props.isLoading();
+
+    if (props.currProject) { 
+      const editedProject = { ...props.currProject };
+      editedProject.name = inputValue
+      props.editProject(editedProject)
+    } else {
+      const newProjectId = uuidv4()
+      const newProject = {id: newProjectId, name: inputValue}
+      props.addProject(newProject)
+    }
+  }
+
+  const renderTitle = () => {
+    if (props.currProject) {
+      return <Title>{props.currProject.name}</Title>
+    } else {
+      return <Title>Add new project</Title>
+    }
+  }
+
+  const renderInput = () => {
+    if (props.currProject) {
+      return <Input type="text" name="name" defaultValue={props.currProject.name}/>
+    } else {
+      return <Input type="text" name="name"/>
+    }
   }
 
   return (
     <Backdrop>
       <Modal>
-        <CloseButton onClick={props.onCloseEditModal}>X</CloseButton>
-        <Title>{ props.title }</Title>
-        <form onSubmit={ submit }>
+        <CloseButton onClick={props.closeEditModal}>X</CloseButton>
+        {renderTitle()}
+        <form onSubmit={submitProject}>
           <Label>Project Label</Label>
-          <Input type="text" name="name" defaultValue={ props.inputValue }/>
+          {renderInput()}
             <Button>SAVE</Button>
         </form>
       </Modal>
     </Backdrop>
   )
 }
 
+EditProject.propTypes = {
+  editProject: PropTypes.func,
+  addProject: PropTypes.func,
+  isLoading: PropTypes.func,
+  closeEditModal: PropTypes.func
+}
+
+const mapStateToProps = state => ({
+  currProject: state.projects.currProject
+})
 
-export default EditProject
\ No newline at end of file
+export default connect(mapStateToProps, { closeEditModal, isLoading, addProject, editProject })(EditProject)
```

Comment:
if you imported * as projectActions you would need to change here as well
```suggestion
export default connect(mapStateToProps, { 
    closeEditModal: projectActions.closeEditModal, 
    .... // complete the rest here
})(EditProject)
```

again, this is just a suggestion. another advantage of this besides name conflicts that come to mind is that you know from where each action creator came from. the downside is that it is more to write